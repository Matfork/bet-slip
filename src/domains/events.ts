import { MarketType } from './market';

export interface EventType {
  id: string;
  name: string;
  markets: MarketType[];
}
