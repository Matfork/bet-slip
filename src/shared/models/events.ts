export interface EventType {
  id: string;
  name: string;
  markets: MarketType[];
}

interface MarketType {
  id: string;
  name: string;
  markets: SelectionType[];
}

interface SelectionType {
  id: string;
  name: string;
  price: number;
}
