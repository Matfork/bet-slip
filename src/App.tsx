import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import EventComponent from 'components/Event/Event';
import { EventsActionCreators } from 'store/actions/action-creators/events.actionCreator';
import HeaderComponent from 'shared/components/Header/Header';
import { RootState } from 'store/reducers';
import { AppInitialState } from 'store/reducers/app.reducer';
import { Backdrop, CircularProgress } from '@material-ui/core';
import './App.scss';

const App: React.FC = () => {
  const { isLoading } = useSelector<RootState, AppInitialState>((state) => state.app);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch({ type: EventsActionCreators.RequestEvents });
  }, []);

  return (
    <div className="App-component">
      <HeaderComponent />

      {isLoading ? (
        <Backdrop open>
          <CircularProgress color="inherit" />
        </Backdrop>
      ) : (
        <div className="App-component__responsive-container">
          <EventComponent />
        </div>
      )}
    </div>
  );
};

export default App;
