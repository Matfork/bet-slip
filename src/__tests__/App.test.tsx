import { screen } from '@testing-library/react';
import { EventsActionCreators } from 'store/actions/action-creators/events.actionCreator';
import App from '../App';
import { initialStateStub } from '../test.stub';
import { customRender } from 'test.utilities';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  ...(jest.requireActual('react-redux') as {}),
  useDispatch: () => mockDispatch,
}));

describe('App Component', () => {
  it('should test snapshot', () => {
    const component = customRender(<App />);
    expect(component.container).toMatchSnapshot();

    expect(mockDispatch).toHaveBeenCalledWith({ type: EventsActionCreators.RequestEvents });
    expect(mockDispatch).toHaveBeenCalledTimes(1);
  });

  it('should display backdrop if data is fetching', () => {
    const component = customRender(<App />, {
      initialState: {
        ...initialStateStub,
        app: { ...initialStateStub.app, isLoading: true },
      },
    });
    const backDropElement = component.container.closest('body')!.querySelector('.MuiBackdrop-root');
    expect(backDropElement).toBeInTheDocument();
  });

  it('should display events if there is data available', () => {
    customRender(<App />);
    const titleEl = screen.queryByText(/Real Madrid vs Barcelona/);
    expect(titleEl).toBeInTheDocument();
  });
});
