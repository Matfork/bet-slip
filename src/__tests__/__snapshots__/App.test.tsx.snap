// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should test snapshot 1`] = `
<div>
  <div
    class="App-component"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary Header-component__container MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root Header-component__title MuiTypography-h6"
        >
          Bet Split
        </h6>
        <button
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root Header-component__hamburger MuiIconButton-colorInherit MuiIconButton-edgeEnd"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <div
      class="App-component__responsive-container"
    >
      <div
        class="Event-component"
      >
        <div
          class="MuiPaper-root MuiCard-root Event-component__card MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            class="MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Real Madrid vs Barcelona
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="Event-component__markets"
            >
              <div
                class="Market-component"
              >
                <div
                  class="Market-component__container"
                >
                  <div
                    class="Market-component__label"
                  >
                    <h6
                      class="MuiTypography-root MuiTypography-h6"
                    >
                      Team to Win
                    </h6>
                  </div>
                  <div
                    class="Market-component__options"
                  >
                    <div
                      class="Selection-component"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            Real Madrid
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            1.23
                          </p>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            Barcelona
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            2.05
                          </p>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root"
                  />
                </div>
                <div
                  class="Market-component__container"
                >
                  <div
                    class="Market-component__label"
                  >
                    <h6
                      class="MuiTypography-root MuiTypography-h6"
                    >
                      Player to Score First
                    </h6>
                  </div>
                  <div
                    class="Market-component__options"
                  >
                    <div
                      class="Selection-component"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            Ronaldo
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            1.15
                          </p>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            Messi
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            1.3
                          </p>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            Bale
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            1.35
                          </p>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiCard-root Event-component__card MuiPaper-outlined MuiPaper-rounded"
        >
          <div
            class="MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
              >
                Atletico Madrid vs Malaga
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root"
          >
            <div
              class="Event-component__markets"
            >
              <div
                class="Market-component"
              >
                <div
                  class="Market-component__container"
                >
                  <div
                    class="Market-component__label"
                  >
                    <h6
                      class="MuiTypography-root MuiTypography-h6"
                    >
                      Team to Win
                    </h6>
                  </div>
                  <div
                    class="Market-component__options"
                  >
                    <div
                      class="Selection-component"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            Atletico
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            1.4
                          </p>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-outlined Selection-component__button  MuiButton-outlinedSizeSmall MuiButton-sizeSmall MuiButton-disableElevation"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            Malaga
                          </p>
                          <p
                            class="MuiTypography-root MuiTypography-body2"
                          >
                            3.05
                          </p>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <hr
                    class="MuiDivider-root"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
